selendroid-server/src/main/java/io/selendroid/server/model/DefaultSelendroidDriverFactory.java

```diff
@@ -4,9 +4,24 @@
 
 public class DefaultSelendroidDriverFactory implements SelendroidDriverFactory {
 
+    private static SelendroidDriver selendroidDriver = null;
+
+    //How about having an enum?
+    private static final String AN_SELENDROID = "selendroid";
+    private static final String AN_UIAUTOMATOR = "UiAutomator";
+
     @Override
-    public SelendroidDriver createSelendroidDriver( ServerInstrumentation androidInstrumentation ) {
-        return new DefaultSelendroidDriver(androidInstrumentation);
+    public SelendroidDriver createSelendroidDriver( ServerInstrumentation androidInstrumentation, String automationName ) {
+
+        if(selendroidDriver == null) {
```

Comment:
Formatting of this code is weird. Android Studio should auto format it correctly.